### PR TITLE
Tests: termination abci

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

small gaps in termination abci, specifically the get_callback_request override.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/skills/termination_abci/tests/test_behaviours.py [hunk 1]
```diff
@@ -488,6 +506,26 @@ def _mock_get_tx_data_request(
...
+            BehaviourTestCase(
+                name="agent could not send message because state != WAITING_MESSAGE",
+                initial_data=_INITIAL_DATA,
+                ok_reqs=[],
+                err_reqs=[
+                    _mock_state_is_not_waiting_message,
+                    _mock_get_service_owner_request,
+                ],
+                expected_logs=[_IS_NOT_WAITING_MESSAGE],
+            ),
```
Comment (Karrenbelt) [inline packages/valory/skills/termination_abci/tests/test_behaviours.py:528]: @0xArdi is this acceptable? These first of these `err_reqs` here are plain mocks, not request mocks. I didn't want to change the `BehaviourTestCase` with additional "pre-mocks". Also the `_mock_get_service_owner_request` is not truly an error mock, just need a mock that will trigger the callback, but not trigger `.complete()`.

Comment (0xArdi) [reply packages/valory/skills/termination_abci/tests/test_behaviours.py:528]: Yeah I think its still clear what is happening. We could think of a better name than ok_reqs, err_reqs for the future.

**Comment by Karrenbelt** [comment]
failure on all OS:
    ```
    2022-12-12T12:01:16.7565122Z E           assert 'dropping message as behaviour has stopped:' in "INFO     aea.test_agent_name.packages.valory.skills.termination_abci:behaviour_utils.py:708 [test_agent_name] Entered in the 'background_behaviour' behaviour\nERROR    aea.packages.valory.protocols.contract_api.message:message.py:469 'data' content is not set.\n"
    2022-12-12T12:01:16.7566525Z E            +  where "INFO     aea.test_agent_name.packages.valory.skills.termination_abci:behaviour_utils.py:708 [test_agent_name] Entered in the 'background_behaviour' behaviour\nERROR    aea.packages.valory.protocols.contract_api.message:message.py:469 'data' content is not set.\n" = <_pytest.logging.LogCaptureFixture object at 0x7ff7b5ef9590>.text
    ```
    
This is noteworthy because there is no such failure observed when running pytest directly
```

pytest ./packages/valory/skills/termination_abci/tests
========================== 23 passed in 0.92s ==========================
```
```
autonomy test by-path ./packages/valory/skills/termination_abci
========================== 1 failed, 22 passed, 3 warnings in 0.80s ==========================
```

and hence added to the existing [issue on dysfunction `test` command](https://github.com/valory-xyz/open-aea/issues/479)

File: scripts/check_doc_links.py [hunk 2]
```diff
@@ -56,6 +56,7 @@
     "https://gateway.autonolas.tech/ipfs/<hash>,",  # non link (400)
     "https://github.com/valory-xyz/open-autonomy/trunk/infrastructure",  # svn link (404)
     "http://host.docker.internal:8545",  # internal (ERR_NAME_NOT_RESOLVED)
+    "https://notes.ethereum.org/@vbuterin/eip-1559-faq",  # 503 Service Temporarily Unavailable
```
Comment (Karrenbelt) [inline scripts/check_doc_links.py:59]: TODO: revert.

This PR will be continued later this week when autonomy test command allows capturing of logs at INFO level
https://github.com/valory-xyz/open-autonomy/blob/adea7fd23e3fef23674ae24776d095ff42579e2c/pytest.ini#L3

File: packages/valory/skills/termination_abci/behaviours.py [hunk 3]
```diff
@@ -500,7 +500,7 @@ def callback_request(
             :return: none
             """
             if self.is_stopped:
-                self.context.logger.debug(
+                self.context.logger.info(
```
Comment (Karrenbelt) [inline packages/valory/skills/termination_abci/behaviours.py:503]: there is an [open issue on open-aea](https://github.com/valory-xyz/open-aea/issues/479) for this

File: packages/valory/skills/abstract_round_abci/tests/conftest.py [hunk 4]
```diff
@@ -39,16 +38,6 @@
...
-@pytest.fixture(scope="module", autouse=True)
-def load_hypothesis_profile() -> Generator:
-    """Fixture to load hypothesis CI settings."""
-    if os.getenv(CI):
-        settings.load_profile(CI)
-    profile = settings.get_profile(settings._current_profile)
-    logging.info(f"Using hypothesis profile from {__file__}:\n{profile}")
-    yield
-
-
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/tests/conftest.py:51]: this clearly did not work, hence I killed it.

File: packages/valory/skills/abstract_round_abci/tests/test_base.py [hunk 5]
```diff
@@ -94,6 +94,8 @@
 )
 
 
+settings.load_profile(os.getenv("CI", "default"))
+
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/tests/test_base.py:98]: this is the alternative approach that I believe does work

File: packages/valory/skills/abstract_round_abci/tests/test_base.py [hunk 6]
```diff
@@ -307,7 +309,6 @@ def test_verify_transaction_negative_case(*_mocks: Any) -> None:
         transaction.verify("")
 
 
-@hypothesis.settings(deadline=2000)
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/tests/test_base.py:310]: I removed all other test specific settings for cleanliness

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**